src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(HashMap<String, Integer> storage, FruitTransaction transaction) {
+        String fruitName = transaction.getFruit();
+        int oldQuantity = storage.get(fruitName);
+        int newQuantity = oldQuantity - transaction.getQuantity();
+        if (newQuantity < 0) {
+            throw new RuntimeException(String.format(
+                    "Error purchasing fruits. Available quantity: %d for %s, "
+                            + "but trying to purchase %d.",
+                    oldQuantity, fruitName, transaction.getQuantity()));
+        }
+        storage.put(fruitName, newQuantity);
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. However, you have implemented this correctly by throwing a RuntimeException if the newQuantity is less than 0, which is good.